src/benchmark/operators/join_aggregate_benchmark.cpp

```diff
@@ -0,0 +1,182 @@
+    auto chunk = zip_table->get_chunk(chunk_index);
+    chunk->finalize();
+    chunk->set_ordered_by(std::make_pair<ColumnID, OrderByMode>(ColumnID{0}, OrderByMode::Ascending));
+    chunk->set_ordered_by(std::make_pair<ColumnID, OrderByMode>(ColumnID{1}, OrderByMode::Ascending));
+  }
+
+  return std::make_shared<TableWrapper>(zip_table);
+}
+
+std::shared_ptr<TableWrapper> create_ages_table(const size_t table_size) {
+
+  auto ages_values = generate_ages(table_size);
+
+  auto ages_table = create_table(table_size, ages_values);
+
+  const auto chunk_count = ages_table->chunk_count();
+  for (ChunkID chunk_index = ChunkID{0}; chunk_index < chunk_count; ++chunk_index) {
+    auto chunk = ages_table->get_chunk(chunk_index);
+    chunk->finalize();
+    chunk->set_ordered_by(std::make_pair<ColumnID, OrderByMode>(ColumnID{0}, OrderByMode::Ascending));
+  }
+
+  return std::make_shared<TableWrapper>(ages_table);
+}
+
+BENCHMARK_F(MicroBenchmarkBasicFixture, BM_Join_Aggregate_Sort)(benchmark::State& state) {
+  _clear_cache();
+
+  auto table_wrapper_left = create_ages_table(TABLE_SIZE);
+  table_wrapper_left->execute();
+  auto table_wrapper_right = create_zip_table(TABLE_SIZE);
+  table_wrapper_right->execute();
+
+  auto operator_join_predicate = OperatorJoinPredicate(std::make_pair<ColumnID, ColumnID>(ColumnID{0}, ColumnID{0}), PredicateCondition::Equals);
+
+
+  auto aggregates = std::vector<std::shared_ptr<AggregateExpression>>{
+    std::static_pointer_cast<AggregateExpression>(avg_(pqp_column_(ColumnID{1}, DataType::Int, false, "b")))};
+
+  std::vector<ColumnID> groupby = {ColumnID{3}};
+
+  auto join = std::make_shared<JoinSortMerge>(table_wrapper_left, table_wrapper_right, JoinMode::Inner,
+                                              operator_join_predicate);
+  join->execute();
+  auto warm_up = std::make_shared<AggregateSort>(join, aggregates, groupby);
+  warm_up->execute();
+
+  for (auto _ : state) {
+    auto join = std::make_shared<JoinSortMerge>(table_wrapper_left, table_wrapper_right, JoinMode::Inner,
+                                                operator_join_predicate);
+    join->execute();
+    auto aggregate_sort = std::make_shared<AggregateSort>(join, aggregates, groupby);
+    aggregate_sort->execute();
+  }
+}
+
+BENCHMARK_F(MicroBenchmarkBasicFixture, BM_Join_Aggregate_Hash)(benchmark::State& state) {
+  _clear_cache();
+
+  auto table_wrapper_left = create_ages_table(TABLE_SIZE);
+  table_wrapper_left->execute();
+  auto table_wrapper_right = create_zip_table(TABLE_SIZE);
+  table_wrapper_right->execute();
+
+  auto operator_join_predicate = OperatorJoinPredicate(std::make_pair<ColumnID, ColumnID>(ColumnID{0}, ColumnID{0}), PredicateCondition::Equals);
+
+  auto aggregates = std::vector<std::shared_ptr<AggregateExpression>>{
+    std::static_pointer_cast<AggregateExpression>(avg_(pqp_column_(ColumnID{1}, DataType::Int, false, "b")))};
```

Comment:
Can we template the function on join_type and agg_type and execute this function (which is currently more or less  duplicated) with all pairs of possible combinations?
The join_benchmark.cpp does it the same way and it's quite short. Or is there a good reason not to template the operators here?

Reply:
Yes, it's a good idea to template and yes, it was pretty simple. Thanks for the hint :)